### PR TITLE
Commentsview

### PR DESCRIPTION
**Pull request by Marcosnr97**
### Summary

- Implemented a method to add new comments to a movie

- Implemented a method to read comments of a movie

### Trello Card
https://trello.com/c/fCSjVEhH

### CODE REVIEW

File: training-net/src/Controllers/MoviesController.cs [hunk 1]
```diff
@@ -142,13 +142,30 @@ public IActionResult Edit(MovieViewModel mvm)
...
-        {
-            var movie = UnitOfWork.MovieRepository.Get(id);
+        {   
+            var movie = UnitOfWork.MovieRepository.GetMovieWithComments(id);
             if (movie == null)
             {
                 return NotFound();
             }
-            return View(new MovieViewModel { ID = movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate, Genre = movie.Genre, Price = movie.Price });
+            //var comments = UnitOfWork.CommentRepository.GetAll().Where(comment => comment.MovieID == movie.ID).Select(comment =>  new CommentViewModel { ID = movie.ID, Text = comment.Text, Date = comment.Date, Rating = comment.Rating, MovieID = movie.ID });
```
Comment (gzamudio) [inline training-net/src/Controllers/MoviesController.cs]: remove this comment if is useless

File: training-net/src/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -142,13 +142,30 @@ public IActionResult Edit(MovieViewModel mvm)
...
+             var comments = new List<CommentViewModel>();
```
Comment (gzamudio) [inline training-net/src/Controllers/MoviesController.cs]: here you can use LINQ Queries instead a foreach and a new List declaration

File: training-net/src/Models/Views/MovieViewModel.cs [hunk 3]
```diff
@@ -22,5 +23,7 @@ public class MovieViewModel
         [DataType(DataType.Currency), Range(1, 100)]   
         public decimal Price { get; set; }
         public string Rating { get; set; }
+        public ICollection<CommentViewModel> Comments { get; set; }
+        public CommentViewModel NewComment {get; set; }
```
Comment (gzamudio) [inline training-net/src/Models/Views/MovieViewModel.cs]: check the spaces

File: training-net/src/Repositories/CommentRepository.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public class CommentRepository : Repository<Comment>, ICommentRepository
+    {
+        public CommentRepository(DbContext context) : base(context)
+        {
+        }
+
+        public DbContext DbContext{ get { return Context as DbContext; } }
+
+        // IEnumerable<Comment> ICommentRepository.GetMovieComments(int id)
```
Comment (gzamudio) [inline training-net/src/Repositories/CommentRepository.cs]: delete this commented method if is useless

File: training-net/src/Repositories/Database/DataBaseContext.cs [hunk 5]
```diff
@@ -12,6 +12,7 @@ public class DataBaseContext : IdentityDbContext<User>
         public DataBaseContext(DbContextOptions<DataBaseContext> options) : base(options) {}
 
         public DbSet<Movie> Movies { get; set; }
+        public DbSet<Comment> Comment { get; set; }
```
Comment (gzamudio) [inline training-net/src/Repositories/Database/DataBaseContext.cs:15]: if the name is Movies, use Comments.

File: training-net/src/Repositories/Interfaces/ICommentRepository.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using MvcMovie.Models;
+using Repositories.Interfaces;
+
+namespace Repositories.Interfaces
+{
+    public interface ICommentRepository : IRepository<Comment>
+    {
+       // IEnumerable<Comment> GetMovieComments(int id);
```
Comment (gzamudio) [inline training-net/src/Repositories/Interfaces/ICommentRepository.cs]: delete this if is useless

File: training-net/src/Controllers/MoviesController.cs [hunk 7]
```diff
@@ -159,7 +176,7 @@ public IActionResult Delete(int id)
             {
                 return NotFound();
             }
-            return View(new MovieViewModel { ID = movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate,Genre = movie.Genre, Price = movie.Price });  
+            return View(new MovieViewModel { ID = (int)movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate,Genre = movie.Genre, Price = movie.Price});
```
Comment (gzamudio) [inline training-net/src/Controllers/MoviesController.cs]: again... i dont understand why you have to cast the movie Id

File: training-net/src/Controllers/MoviesController.cs [hunk 8]
```diff
@@ -142,13 +142,30 @@ public IActionResult Edit(MovieViewModel mvm)
...
+             foreach(var comment in movie.Comments)
+             {
+                 var com = new CommentViewModel { ID = comment.ID, Date = comment.Date, Text = comment.Text, Rating = comment.Rating };
+                 comments.Add(com);
+             }
+            return View(new MovieViewModel { ID = movie.ID, Title = movie.Title, ReleaseDate = movie.ReleaseDate, Genre = movie.Genre, Price = movie.Price, Comments = comments });
+        }
+
+        [HttpPost]
+        public IActionResult AddComment(MovieViewModel mvm)
```
Comment (gzamudio) [inline training-net/src/Controllers/MoviesController.cs]: although this is working fine.. it's not what you have to do. we need to add a comment to a movie via AJAX, and here is added by a form. we need that the app don't refresh all the page when a comment is added, only add a comment vía AJAX to the DB and show it in the view. research about this in Google and if you cant solve it let me know

**Review by gzamudio — DISMISSED** [review]

**Review by gussiciliano — APPROVED** [review]

**Pull request merged**